src/RedisCache.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+use MediaWiki\MediaWikiServices;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Curse Inc.
+ * Redis Cache
+ *
+ * @author        Alexia E. Smith
+ * @copyright    (c) 2015 Curse Inc.
+ * @license        GNU General Public License v3.0 only
+ * @package        RedisCache
+ * @link        https://gitlab.com/hydrawiki
+ *
+ **/
+class RedisCache {
+	/**
+	 * Up and ready server connections.
+	 */
+	private array $servers = [];
+
+	/**
+	 * Last exception error.
+	 */
+	private string $lastError = '';
+
+	public function __construct( private ?array $redisServers, private LoggerInterface $logger ) { }
+
+	/**
+	 * @param ?string $group [Optional] Server group key.
+	 *                    Example: 'cache' would look up $wgRedisServers['cached']
+	 *                    Default: Uses the first index of $wgRedisServers.
+	 * @param ?array $options [Optional] Additional options, will merge and overwrite default options.
+	 *                    - connectTimeout : The timeout for new connections, in seconds.
+	 *                      Optional, default is 1 second.
+	 *                    - persistent     : Set this to true to allow connections to persist across
+	 *                      multiple web requests. False by default.
+	 *                    - password       : The authentication password, will be sent to Redis in clear text.
+	 *                      Optional, if it is unspecified, no AUTH command will be sent.
+	 *                    - serializer     : Set to "php", "igbinary", or "none". Default is "php".
+	 * @param    ?boolean $newConnection [Optional] Force a new connection, useful when forking processes.
+	 * @return RedisConnRef|null RedisConnRef or null on failure.
+	 * @throws MWException
+	 */
+	public function getConnection(
+		?string $group = null,
+		?array $options = [],
+		?bool $newConnection = false
+	): ?RedisConnRef {
+		if ( !extension_loaded( 'redis' ) ) {
+			throw new MWException(
+				__METHOD__ .
+				" - The PHP Redis extension is not available.  Please enable it on the server to use RedisCache."
+			);
+		}
```

Comment:
`MWException` just got deprecated (ref https://phabricator.wikimedia.org/T321683#8347775, https://phabricator.wikimedia.org/T86704), so we should use e.g. `Wikimedia\Assert` instead:
```php
Assert::precondition(
extension_loaded( 'redis' ),
__METHOD__ .
				" - The PHP Redis extension is not available.  Please enable it on the server to use RedisCache."
);
```